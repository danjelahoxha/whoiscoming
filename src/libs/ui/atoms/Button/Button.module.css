.button {
  @apply inline-flex prose prose-button py-16 px-40;
  @apply border rounded-lg outline-0;
  @apply transition-all duration-300;
}
