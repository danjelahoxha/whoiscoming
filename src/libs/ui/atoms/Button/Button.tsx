import { classNames } from '@whoiscoming-ui/utilities';

import styles from './Button.module.css';

import type { CSSProperties, ReactNode } from 'react';

type ButtonProps = {
  icon?: ReactNode;
  dataTestId?: string;
  style?: CSSProperties;
  className?: string;
  children: ReactNode;
};

export const Button = ({
  icon,
  dataTestId,
  style,
  className,
  children,
}: ButtonProps) => (
  <button
    data-testid={dataTestId}
    style={style}
    className={classNames([styles.button, className || ''])}
  >
    {icon && <span className='mr-20'>{icon}</span>}
    {children}
  </button>
);
