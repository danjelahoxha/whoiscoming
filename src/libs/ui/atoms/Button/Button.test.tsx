import { render } from '@testing-library/react';

import { Button } from './Button';

describe('<Button>', () => {
  it("should match is's snapshot", () => {
    const { container, rerender } = render(<Button>Hello</Button>);
    expect(container).toMatchSnapshot('default');

    rerender(<Button outline>Hello</Button>);
    expect(container).toMatchSnapshot('outline');

    rerender(<Button icon='i'>Hello</Button>);
    expect(container).toMatchSnapshot('icon default');

    rerender(
      <Button icon='i' outline>
        Hello
      </Button>
    );
    expect(container).toMatchSnapshot('icon outline');
  });
});
