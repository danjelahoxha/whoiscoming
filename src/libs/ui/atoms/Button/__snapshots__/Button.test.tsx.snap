// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> should match is's snapshot: default 1`] = `
<div>
  <button
    class="button default"
  >
    Hello
  </button>
</div>
`;

exports[`<Button> should match is's snapshot: icon default 1`] = `
<div>
  <button
    class="button default"
  >
    <span
      class="mr-20"
    >
      i
    </span>
    Hello
  </button>
</div>
`;

exports[`<Button> should match is's snapshot: icon outline 1`] = `
<div>
  <button
    class="button outline"
  >
    <span
      class="mr-20"
    >
      i
    </span>
    Hello
  </button>
</div>
`;

exports[`<Button> should match is's snapshot: outline 1`] = `
<div>
  <button
    class="button outline"
  >
    Hello
  </button>
</div>
`;
